src/modules/projectSearchContext.js

```diff
@@ -30,6 +29,95 @@ class ProjectSearchContext {
             body: JSON.stringify(searchRequest)
         }).then(this.sessionManager.toJson);
     };
+
+    /**
+     * Search for traces in a project and map every trace to the callback.
+     * 
+     * @param {string|object} request The query as an HQL query string, or the full REST request for the /search
+     * @param {function} callback The callback that receives a trace result object
+     * @returns 
+     */
+    tracesMap = (request = '', callback) => {
```

Comment:
traceMap instead of tracesMap

Reply:
Having one method `traces()` with an optional callback feels more javascripty. Or have `traces()` for the version with callback and `tracesSync()` for the version that returns an array (just like the `fs` standard library of node).

Reply:
OK. Created a single traces() method